lib/matplotlib/collections.py

```diff
@@ -189,7 +189,7 @@ def get_datalim(self, transData):
         paths = self.get_paths()
 
         if not transform.is_affine:
-            paths = [transform.transform_path_non_affine(p) for p in paths]
+            paths = self._non_affine_transform_paths(paths, transform)
```

Comment:
We've gone here from calling `transform_path_non_affine` to (effectively) `transform_non_affine`. The former returns a new `Path`, which based on my reading would seem to offer possibly interpolate points along the path in the transformed coordinate system. The latter will not do this. Now, does anyone know if this is important?

Is it possible to instead combine the individual Paths into one big path and use `transform_path_non_affine`?

Reply:
Oh, hummm.  OK< I just looked at the default `transform_path_non_affine` but that can be overridden, so I think you're right that we need to call that instead....

Reply:
I looked at the default as well, and the line I'm concerned we're skipping is:
```python
return Path._fast_from_codes_and_verts(x, path.codes,                                      
                 {'interpolation_steps': path._interpolation_steps,                                 
                  'should_simplify': path.should_simplify})   
```
Your current implementation calls `self.transform_non_affine(vertices)`, just like the default `transform_path_non_affine`, but it reuses the existing `Path` instances.

Reply:
Ok, I may be over paranoid. It looks like `_fast_from_codes_and_verts` just sets the interpolation steps, it doesn't do any actual interpolation at that time.

Reply:
https://github.com/matplotlib/matplotlib/blob/b0786438b90ccac2f397af994fee7a9d09f7673c/lib/matplotlib/projections/geo.py#L260-L265 Looks to me like the path changes length

Reply:
So, its possible to come up w/ an algorithm that does this if `transform_path_non_affine` uses only `path.interpolated` to get the higher-res path.  But, I don't see any reason that an arbitrary user-supplied transform would be restricted to doing that method of interpolation.   

Given that ambiguity, I think its impossible to come up with a foolproof algorithm for this.  I suppose the special case of no interpolation could be checked for and would work.  

I'll point out that @astrofrog's fix is obviously counting on the non-affine path transform not doing any interpolation - i.e. its doing linear dot connecting after the tranform.   Certainly that'd be a disaster for a geographic non-affine transform (i.e. you want a straight line in lat/lon to be curved in the projection), but maybe is fine for astropy?

Reply:
@jklymak - ah I see - in our case we write the transforms we are using and we don't make use of interpolation for now so things should be fine. It would be pretty rare for the resolution of the contour map to be so low that there would be significant distortions between vertices of the path.